src/cupbearer/detectors/statistical/mahalanobis_detector.py

```diff
@@ -6,11 +6,12 @@
 )
 
 
-def _pinv(C, rcond):
+def _pinv(C, rcond, dtype=torch.float64):
     # Workaround for pinv not being supported on MPS
+    og_dtype = C.dtype
```

Comment:
```suggestion
```
Nit: Personally I find C.dtype to be clearer with what it means. But, is fine as is as well.

Reply:
`input_dtype` is also slightly clearer to me if you think `C.dtype` could be confusing with previous casting and so on. 

My confusion with og is probably that:
1. I haven't really seen that shortening of original much before
2. Inverse was just calculated so it's not the original dtype of the inverse (`input_dtype` doesn't really improve for this point)